tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc

```diff
@@ -270,43 +270,28 @@ LogicalResult ConvertTFSignOp::matchAndRewrite(
 LogicalResult ConvertTFSinOp::matchAndRewrite(Operation* op,
                                               PatternRewriter& rewriter) const {
   auto tf_sin_op = cast<TF::SinOp>(op);
-  ShapedType output_type =
-      mlir::cast<ShapedType>(tf_sin_op.getResult().getType());
 
-  std::optional<Value> result =
-      convertSinOp(rewriter, op, tf_sin_op.getX(), output_type);
-  if (!result) return failure();
+  ShapedType output_type = dyn_cast<TensorType>(tf_sin_op.getResult().getType());
```

Comment:
OOC why cast to TensorType and assign to ShapedType?

Reply:
Also this doesn't seem possible: it would fail verification.

Reply:
thanks Jacques! it should be bug, updated in the latest PS